Show feedback when no comment is provided.

staubina opened the pull request:
TNL-6016

## [TNL-6016](https://openedx.atlassian.net/browse/TNL-6016)

### Description

Peer review grades missing when feedback is optional.

- Removed/moved the logic to allow the grade to show when feedback is not available.

### Sandbox

- [x] Build a sandbox for your branch and add a link here
https://staubina-ora2.sandbox.edx.org


### Testing
- [x] Unit, integration, acceptance tests as appropriate

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @cahrens 
- [x] Code review: @andy-armstrong  
- [ ] Product review: @sstack22 

FYI @catong (The screenshots need to be updated from a previous change.

### Post-review
- [x] Rebase and squash commits

staubina commented:
@andy-armstrong You were the last to edit these files.  Will you please take a look at this PR?

cahrens commented on openassessment/templates/openassessmentblock/grade/oa_assessment_feedback.html:
```diff
@@ -4,21 +4,21 @@
+                                {% blocktrans with title=individual_assessment.title grade=individual_assessment.option.label start_tag='<span class="answer__source__value">'|safe end_tag="</span>"|safe %}
+                                    {{ start_tag }}{{ title }}{{ end_tag }} - {{ grade }}
+                                {% endblocktrans %}
+                            {% else %}
```
Perhaps this else should also check that individual_assessment.feedback exists to avoid the "empty header" issue?

staubina commented:
@cahrens I have updated the PR and the sandbox with a small fix.

cahrens commented on openassessment/templates/openassessmentblock/grade/oa_assessment_feedback.html:
```diff
@@ -4,7 +4,6 @@
             {% if assessment.individual_assessments %}
                 {% for individual_assessment in assessment.individual_assessments %}
-                    {% if individual_assessment.feedback %}
                         <div class="answer_title">
```
Nit: fix indenting level for this first section.

cahrens commented:
Looking good, just a couple of nits.

staubina commented:
@cahrens I did not realize you were online yet, I was testing some changes to correct an issue discovered by the acceptance tests.  I will let you know when it is all set and also when I make your recommended changes.

staubina commented:
Good example ORA problem.
https://staubina-ora2.sandbox.edx.org/courses/course-v1:edx+ORA203+course/courseware/a4dfec19cf9b4a6fb5b18be6ccd9cecc/417e47b2663a4f79b62dba20b21628c8/

staubina commented:
@cahrens I corrected the issues causing acceptance test failures and updated per your comments.  Please take another look when you have time.

cahrens commented on openassessment/templates/openassessmentblock/grade/oa_assessment_feedback.html:
```diff
@@ -1,41 +1,43 @@
-            {% else %}
+          {% if assessment.individual_assessments %}
+            {% for individual_assessment in assessment.individual_assessments %}
+              {% if grade != '' %}
```
When is the grade empty? If it is empty, does it makes sense to show the feedback (as there will be no title)?

cahrens commented:
The peer comments are showing up in the staff column in the "Content" section.

![image](https://cloud.githubusercontent.com/assets/484484/21149374/4876ac58-c129-11e6-957e-7f1ce67fd1e8.png)

staubina commented:
@cahrens @andy-armstrong Please hold off on reviewing this PR.  I am re working things to deal with the layout issues @cahrens pointed out.

staubina commented:
Per comments from @cahrens I have changed the approach used to show this information.  The scores and feedback now render in one list element for the entire assessment type.  So for example all the peer assessments will be in the same li element and the same Column, due to the styling on the li elements.

@cahrens and/or @andy-armstrong , I am going to wait for your feedback before making any further changes to tests or UI.

cahrens commented:
I think you intended for these all to have the same classname, but there is a typo:

![image](https://cloud.githubusercontent.com/assets/484484/21163920/d39920c4-c163-11e6-9512-052d237d666f.png)

That being said, I'm not sure it makes sense for them all to have the same classname. Let's talk tomorrow.

staubina commented:
@cahrens Yup they were pre existing... I think I will need to do some pruning.  I have considered merging the templates in to one, but I am not sure that is needed.

I will look for references and delete the div if it is not needed.

staubina commented:
@cahrens I removed the answer__title and answer_title divs and changed some spans to divs.  The most recent commit has these changes.  I have also updated the Sandbox. 

FYI @andy-armstrong

staubina commented:
@andy-armstrong I am working on an acceptance test update now.

cahrens commented:
👍  Just let me know when the test is available for review.

staubina commented:
@cahrens @andy-armstrong I was finally able to correct all test failures and add to an existing acceptance test.  Let me know if you think I should add more.

cahrens commented on test/acceptance/pages.py:
```diff
@@ -553,6 +553,27 @@ def grade_entry(self, question, column):
+        self.wait_for_element_visibility(
+            self._bounded_selector('.question--{} .answer .answer__source__individual'.format(question + 1)),
+            "Individual peer entry was present",
+            2
```
What is the "2" here (time to wait?)? What was the default?

cahrens commented on test/acceptance/pages.py:
```diff
@@ -553,6 +553,27 @@ def grade_entry(self, question, column):
+
+        """
+        self.wait_for_element_visibility(
+            self._bounded_selector('.question--{} .answer .answer__source__individual'.format(question + 1)),
```
Nit: define the selector as a variable in this method.

cahrens commented on test/acceptance/tests.py:
```diff
@@ -997,6 +1010,10 @@ def test_staff_override_at_end(self):
         ])
 
+        self.verify_individual_entries([
+            (0, 0, u"PEER 1"),
```
Why are these all "PEER 1"?

cahrens commented on test/acceptance/pages.py:
```diff
@@ -553,6 +553,27 @@ def grade_entry(self, question, column):
 
         return source.text.strip(), value.text.strip()
 
+    def get_individual_entry(self, question, row):
```
Can you just return this information in the existing grade_entry method? I'd also like the value to be returned in addition to the peer "name".

staubina commented:
@cahrens @andy-armstrong I have updated the test per the PR comments.

cahrens commented:
👍 Thanks-- I think your last changes simplified the tests quite a bit. If you want to keep the unreferenced classes, that's fine-- I guess it provides a parallel structure to the other grade types.

Merge it!

staubina commented:
@sstack22 Once you review I will be able to merge this PR.

sstack22 commented:
Looks good to me! CC: @JAAkana

catong commented:
@staubina Looks great! Reflecting these changes in doc will lag behind a bit, but DOC-3533 is created for the doc update. (FYI @sstack22 )
